src/rt/msvc.c

```diff
@@ -156,3 +160,55 @@ int  _msvc_fileno(FILE* stream)
 {
     return stream->_file;
 }
+
+
+
+/*** math ***/
+
+// The x86 runtime lacks most 32-bit float *f math functions.
+// Provide alternate implementations using the 64-bit double version.
+#if defined _M_IX86
+
+#define IMPL(baseName) \
```

Comment:
I'd recommend to use more descriptive names for these macros, e.g. `IMPL_ALT_MATHFNF`.